### DISCUSSION TITLE
How do I get callbacks from the webview?

### DISCUSSION THREAD

**Discussion by jamesrp13** [Voltstro-Studios/UnityWebBrowser#280] [Q&A] [answered]
We have an existing mobile SDK that we're adding desktop support to. We hope to use UnityWebBrowser to support Windows. The architecture of our SDK is such that we send raw JS to the browser (using `ExecuteJS`) and then need to be able to receive a callback with the response. Does anyone know how to register for this kind of callback? In the library we're using on mobile it's a field in the webview's constructor.

I'm still trying to wrap my head around how UnityWebBrowser works. I've successfully called `ExecuteJS` to pass the correct input but can anyone point me in the right direction for how to receive a callback from there?

**Comment by Voltstro** [comment]
Currently there is no way for `ExecuteJS` to return anything. CEF's `ExecuteJavaScript` does not return, since if was to return, it would mean blocking the thread.

In the upcoming UWB release, there is a new feature called JS Methods, which does allow callbacks in the sense that the JS code can call a .NET method.

**Reply by jamesrp13** [reply]
Ah that’s a bummer. Is there a timeline associated with that release?

**Reply by jamesrp13** [reply]
The .NET callback seems to be exactly what we need. That’s how the macOS library we’re using functions at the moment - we provide a C# function for it to call whenever it has messages to send back.

**Reply by Voltstro** [reply, marked as answer]
The 2.1.0 release is out now. It includes the new [JS Methods feature](https://projects.voltstro.dev/UnityWebBrowser/latest/articles/user/javascript/#js-methods).

**Reply by jamesrp13** [reply]
Amazing, thank you! I’ll take a look!
